ecl/hqlcpp/hqlcpp.cpp

```diff
@@ -9607,6 +9610,41 @@ void HqlCppTranslator::doBuildExprFailCode(BuildCtx & ctx, IHqlExpression * expr
     }
 }
 
+void HqlCppTranslator::doBuildExprMatchedInJoin(BuildCtx & ctx, IHqlExpression * expr, CHqlBoundExpr & tgt)
+{
+    HqlExprAssociation * matchedMarker = ctx.queryMatchExpr(activeJoinTransformFlagsMarkerExpr);
+    if (!matchedMarker)
+    {
+        if (!buildExprInCorrectContext(ctx, expr, tgt, false))
+            throwError(HQLERR_MatchedContextNotFound);
+        return;
+    }
+
+    __int64 mask = 0;
```

Comment:
I do understand why you want it associated in associateLocalJoinTransformFlags, just something in the details I need to tie it all together.